lib/s3_helper.rb

```diff
@@ -281,7 +281,12 @@ def download_file(filename, destination_path, failure_message = nil)
   end
 
   def s3_client
-    @s3_client ||= Aws::S3::Client.new(@s3_options)
+    @s3_client ||= init_aws_s3_client
+  end
+
+  def stub_client_responses!
+    raise "This method is only allowed to be used in the testing environment" if !Rails.env.test?
+    @s3_client = init_aws_s3_client(stub_responses: true)
   end
```

Comment:
Actually I think there is no need to do this here, you can do something like this in tests instead
```
client = Aws::S3::Client.new(stub_responses: true)
client.stub_responses(xyz)
Aws::S3::Client.stubs(:new).returns(client)
```

Reply:
The problem is that this unnecessarily exposes the internals of `S3Helper` and when a person is initializing `Aws::S3::Client` manually, they are not initializing it with the necessary options which `S3Helper` will initialize the client with.